version: "3.9"

services:
  rabbitmq:
    image: rabbitmq:3-management
    environment:
      RABBITMQ_DEFAULT_USER: ${RABBITMQ_USERNAME}
      RABBITMQ_DEFAULT_PASS: ${RABBITMQ_PASSWORD}
    networks:
      - proxy
    volumes:
      - rabbitmq_data:/var/lib/rabbitmq
    deploy:
      placement:
        constraints: [node.labels.infra == true ]
      restart_policy:
        condition: on-failure
      labels:
        - traefik.enable=true
        - traefik.http.routers.rabbitmq-ui.rule=Host(`rabbitmq-ui.zcorp.ovh`)
        - traefik.http.routers.rabbitmq-ui.entrypoints=websecure
        - traefik.http.routers.rabbitmq-ui.tls.certresolver=myresolver
        - traefik.http.services.rabbitmq-ui.loadbalancer.server.port=15672
        - traefik.tcp.routers.rabbitmq-amqps.rule=HostSNI(`rabbitmq-server.zcorp.ovh`)
        - traefik.tcp.routers.rabbitmq-amqps.entrypoints=amqps
        - traefik.tcp.routers.rabbitmq-amqps.tls.certresolver=myresolver
        - traefik.tcp.routers.rabbitmq-ampqs.service=service-rabbitmq-ampqs
        - traefik.tcp.services.service-rabbitmq-ampqs.loadbalancer.server.port=5672

networks:
  proxy:
    external: true

volumes:
  rabbitmq_data:
